core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3695,13 +3695,21 @@ KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType&) {
   return true;
 }
 
+#if defined(KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK)
 template <unsigned R, class MapType, class iType, class... Args>
 KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType& map,
                                                         const iType& i,
                                                         Args... args) {
   return (size_t(i) < map.extent(R)) &&
          view_verify_operator_bounds<R + 1>(map, args...);
 }
+#else
+template <unsigned R, class MapType, class... Args>
+KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType& /*map*/,
+                                                        Args... /*args*/) {
+  return true;
+}
+#endif
```

Comment:
I am not super happy with this change. Essentially it means for every data access the compiler has to inline/optimize away one more variadic function template call.